src/AutoRest.CSharp/Common/Input/Source/SourceInputModel.cs

```diff
@@ -56,6 +78,48 @@ public ModelTypeMapping CreateForModel(INamedTypeSymbol? symbol)
             return new ModelTypeMapping(_modelAttribute, _schemaMemberNameAttribute, symbol);
         }
 
+        public IMethodSymbol? FindForMethod(string name, IEnumerable<string> parameters)
+        {
+            var methods = _methodSet.Where(m => m.Name == name).ToArray();
+            if (methods.Length == 0)
+            {
+                return null;
+            }
+            else if (methods.Length == 1)
+            {
+                return methods.First();
```

Comment:
Why immediately return when there is only one method with the same method name? the method may don't have the same parameters.

Reply:
Because this is enough. We cannot define two methods with same name. What's more, we cannot compare the types technically.